tests/codemods/semgrep/test_semgrep_nan_injection.py

```diff
@@ -0,0 +1,71 @@
+import json
+
+from codemodder.codemods.test import BaseSASTCodemodTest
+from core_codemods.semgrep.semgrep_nan_injection import SemgrepNanInjection
+
+
+class TestSemgrepNanInjection(BaseSASTCodemodTest):
```

Comment:
I think we need some additional cases where the `float(potential_nan)` is used in a context other than simple assignment.

I also think we should add test cases for the other type casts that are identified by this Semgrep rule.